cpp/src/arrow/util/value_parsing_test.cc

```diff
@@ -120,6 +120,14 @@ TEST(StringConversion, ToInt8) {
   AssertConversionFails<Int8Type>("-");
   AssertConversionFails<Int8Type>("0.0");
   AssertConversionFails<Int8Type>("e");
+
+  // Hex
+  AssertConversion<Int8Type>("0x0", 0);
+  AssertConversion<Int8Type>("0x1A", 26);
+  AssertConversion<Int8Type>("0xb", 11);
+  AssertConversion<Int8Type>("0x7F", 127);
```

Comment:
What happens if I pass `0x80` or higher? Does it work?

Reply:
added unit test for this case